### PR TITLE
12.0 auth company signup

### PR DESCRIPTION
**Pull request by houssine78**
This module bring to possible for a user to signup as a company. It will also have the possibility to introduce a VAT number which is not mandatory.

### CODE REVIEW

File: auth_company_signup/controllers/main.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
```
Comment (vvrossem) [inline auth_company_signup/controllers/main.py]: to remove

File: auth_company_signup/controllers/main.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+        res_lan_obj = request.env['res.lang']
+
+        supported_langs = [
+            l['code'] for l in res_lan_obj.sudo().search_read([], ['code'])
+        ]
+        if request.lang in supported_langs:
+            values['lang'] = request.lang
+        self._signup_with_values(qcontext.get('token'), values)
+        request.env.cr.commit()
```
Comment (vvrossem) [inline auth_company_signup/controllers/main.py]: missing EOL

File: auth_company_signup/static/src/js/auth_signup.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+			var val = $("input[name='is_company']:checked").val();
+			console.log(val);
+			if(val == "on") {
+				$("div[name='vat_div']").show('quick');
+			} else {
+				$("div[name='vat_div']").hide('quick');
+			}
+		});
+	});
+});
```
Comment (vvrossem) [inline auth_company_signup/static/src/js/auth_signup.js]: missing EOL

File: auth_company_signup/static/src/js/auth_signup.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+odoo.define('auth_company_signup.auth_signup', function (require) {
+	"use strict";
+
+	$(document).ready(function () {
+		$("input[name='is_company']").click(function(ev) {
+			var val = $("input[name='is_company']:checked").val();
+			console.log(val);
```
Comment (vvrossem) [inline auth_company_signup/static/src/js/auth_signup.js]: remove console.log()

File: auth_company_signup/views/auth_signup_assets.xml [hunk 5]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <template id="assets_frontend" inherit_id="web.assets_frontend" name="Auth company signup">
+        <xpath expr="." position="inside">
+              <script src="/auth_company_signup/static/src/js/auth_signup.js" type="text/javascript"/>
+        </xpath>
+    </template>
+</odoo>
```
Comment (vvrossem) [inline auth_company_signup/views/auth_signup_assets.xml]: missing EOL

File: auth_company_signup/views/auth_signup_template.xml [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+            
+        </xpath>
+        <xpath expr="//input[@name='name']/.." position="after">
+            <div name ="vat_div" class="form-group field-name" style="display:none">
+                <label for="vat">VAT Number</label>
+                <input type="text" name="vat" t-att-value="vat" id="vat" class="form-control form-control-sm" />
+            </div>
+        </xpath>
+     </template>
+</odoo>
```
Comment (vvrossem) [inline auth_company_signup/views/auth_signup_template.xml]: missing EOL

**Review by vvrossem — CHANGES_REQUESTED** [review]

File: auth_company_signup/controllers/main.py (hunk 1 above)
Comment (houssine78) [reply auth_company_signup/controllers/main.py]: no we should keep it as the code is from the Odoo module auth_signup

**Comment by houssine78** [comment]
@vvrossem done

**Comment by vvrossem** [comment]
@houssine78, could you make travis happy again (only for [auth_company_signup](https://travis-ci.org/github/coopiteasy/addons/jobs/769581937#L373)): 
- mostly `W7910(wrong-tabs-instead-of-spaces)`
- missing readme

:pray:

**Pull request merged**